interchaintest/chain_upgrade_test.go

```diff
@@ -2,28 +2,55 @@ package interchaintest
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
 	"testing"
 	"time"
 
-	"github.com/strangelove-ventures/interchaintest/v4"
-	"github.com/strangelove-ventures/interchaintest/v4/chain/cosmos"
-	"github.com/strangelove-ventures/interchaintest/v4/ibc"
-	"github.com/strangelove-ventures/interchaintest/v4/testutil"
+	"github.com/icza/dyno"
+	"github.com/strangelove-ventures/interchaintest/v7"
+	"github.com/strangelove-ventures/interchaintest/v7/chain/cosmos"
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/strangelove-ventures/interchaintest/v7/testutil"
 	"github.com/stretchr/testify/require"
 	"go.uber.org/zap/zaptest"
 )
 
 const (
-	haltHeightDelta    = uint64(7) // will propose upgrade this many blocks in the future
+	haltHeightDelta    = uint64(15) // will propose upgrade this many blocks in the future
 	blocksAfterUpgrade = uint64(10)
-	votingPeriod       = "10s"
-	maxDepositPeriod   = "10s"
 )
 
 func TestBasicJunoUpgrade(t *testing.T) {
 	repo, version := GetDockerImageInfo()
+	startVersion := "v14.1.0"
 	upgradeName := "v15"
-	CosmosChainUpgradeTest(t, "juno", "v14.1.0", version, repo, upgradeName)
+	CosmosChainUpgradeTest(t, "juno", startVersion, version, repo, upgradeName)
+}
+
+// With us upgrading from v45 to v47, we need to modify the params in the v45 style here & only here.
+// In the future this will be removed to use just `modifyGenesisShortProposals`
+func modifySDKv45GenesisShortProposals(votingPeriod string, maxDepositPeriod string) func(ibc.ChainConfig, []byte) ([]byte, error) {
```

Comment:
Move my localjuno generalized version of this function (with the KV list). Should upstream that to ictest main as well (under the cosmos type)